index.js

```diff
@@ -43,12 +154,20 @@ module.exports = {
 		}
 
 		for (let i = 0; i < pathArray.length; i++) {
-			object = object[pathArray[i]];
+			const key = pathArray[i];
+			const index = Number.parseInt(key, 10);
+
+			// Disallow string indexes
+			if (!Number.isInteger(key) && Array.isArray(object) && !Number.isNaN(index) && object[index] === object[key]) {
```

Comment:
Because JavaScript automatically coerces `object["1"]` to `object[1]` we need to detect that and override that default behaviour

Reply:
This would also be good as a code comment.